src/toniarts/openkeeper/tools/modelviewer/ModelViewer.java

```diff
@@ -582,11 +580,12 @@ private void setupModel(final Node spat, boolean isMap) {
         spat.depthFirstTraversal(new SceneGraphVisitor() {
             @Override
             public void visit(Spatial spatial) {
-                AnimControl animControl = (AnimControl) spatial.getControl(AnimControl.class);
-                if (animControl != null) {
-                    AnimChannel channel = animControl.createChannel();
-                    channel.setAnim("anim");
-                    AnimationLoader.setLoopModeOnChannel(spatial, channel);
+                var animComposer = spatial.getControl(AnimComposer.class);
+                if (animComposer != null) {
+                    animComposer.setCurrentAction("anim");
+                    // var channel = animControl.getAction(NODE_NAME).createChannel();
+                    // channel.setAnim("anim");
+                    // AnimationLoader.setLoopModeOnChannel(spatial, channel);
```

Comment:
?

Reply:
> The only problem with it that it doesn't natively support the loop modes. And it can't be (at the time at least) extended to support them as there is a bug/feature in there that prevents it.

I see, maybe it's fixed by now?

Reply:
https://github.com/jMonkeyEngine/jmonkeyengine/pull/1038 this at least was never merged to the engine... I was a bit lazy on this, should have pursued it further... But maybe there is some way. I just stopped because I wanted to wait for this. Never looked elsewhere.

Reply:
I see, what about https://github.com/jMonkeyEngine/jmonkeyengine/pull/1038#issuecomment-792282131?

Reply:
I think the problem was something that while we could play it backwards on the second loop, we couldn't catch that anymore ending (to have it again looping but this time forward again). It was some time ago. ModelViewer is anyway the easiest to catch these. Some Imp work animations, if I remember correctly, had different loop modes to test on.